crates/aptos-runtimes/src/thread_manager.rs

```diff
@@ -0,0 +1,218 @@
+        THREAD_CONFIG_STRATEGY
+            .set(strategy)
+            .expect("ThreadConfigStrategy can only be set once.");
+    }
+
+    pub fn get_exe_runtime(&self) -> Handle {
+        self.exe_threads.runtime.handle().clone()
+    }
+
+    pub fn get_exe_cpu_pool(&self) -> &ThreadPool {
+        &self.exe_threads.cpu_pool
+    }
+
+    pub fn get_non_exe_runtime(&self) -> Handle {
+        self.non_exe_threads.runtime.handle().clone()
+    }
+
+    pub fn get_non_exe_cpu_pool(&self) -> &ThreadPool {
+        &self.non_exe_threads.cpu_pool
+    }
+
+    pub fn get_io_pool(&self) -> &ThreadPool {
+        &self.io_threads
+    }
+
+    fn new(strategy: ThreadConfigStrategy) -> Self {
+        match strategy {
+            ThreadConfigStrategy::DefaultStrategy => {
+                let exe_threads = Self::create_threads(ThreadsConfigs::default(), "exe");
+                let non_exe_threads = Self::create_threads(ThreadsConfigs::default(), "non_exe");
+                let io_threads =
+                    Self::create_io_threads(ThreadsConfig::new_without_start_hook(64), "io");
+                Self {
+                    exe_threads,
+                    non_exe_threads,
+                    io_threads,
+                }
+            },
+
+            #[cfg(target_os = "linux")]
+            ThreadConfigStrategy::PinExeThreadsToCores(num_exe_cpu) => {
+                let core_ids = core_affinity::get_core_ids().unwrap();
+                assert!(core_ids.len() > num_exe_cpu);
+
+                let mut exe_cpu_set = Self::new_cpu_set();
+                let mut non_exe_cpu_set = Self::new_cpu_set();
+                for core_id in core_ids.iter().take(num_exe_cpu) {
+                    unsafe { CPU_SET(core_id.id, &mut exe_cpu_set) };
+                }
+                for core_id in core_ids.iter().skip(num_exe_cpu) {
+                    unsafe { CPU_SET(core_id.id, &mut non_exe_cpu_set) };
+                }
+
+                let exe_threads = Self::create_threads(
+                    ThreadsConfigs {
+                        tokio_config: ThreadsConfig::new(
+                            num_exe_cpu,
+                            Self::pin_cpu_set(exe_cpu_set),
+                        ),
+                        rayon_config: ThreadsConfig::new(
+                            num_exe_cpu,
+                            Self::pin_cpu_set(exe_cpu_set),
+                        ),
+                    },
+                    "exe",
+                );
+
+                let non_exe_threads = Self::create_threads(
+                    ThreadsConfigs {
+                        tokio_config: ThreadsConfig::new(
+                            core_ids.len() - num_exe_cpu,
+                            Self::pin_cpu_set(exe_cpu_set),
+                        ),
+                        rayon_config: ThreadsConfig::new(
+                            core_ids.len() - num_exe_cpu,
+                            Self::pin_cpu_set(exe_cpu_set),
```

Comment:
`non_exe_cpu_set` ?

Reply:
Good catch, copied the wrong line when doing refactoring.